code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -2077,4 +2083,149 @@
 		to_chat(src, "<span class='xenodanger'>You feel your acid glands refill. You can spray acid again.</span>")
 		for(var/X in actions)
 			var/datum/action/A = X
-			A.update_button_icon()
\ No newline at end of file
+			A.update_button_icon()
+
+
+
+//Defiler abilities
+/mob/living/carbon/Xenomorph/Defiler/proc/emit_neurogas()
+
+	if(!check_state()) return
+
+	if(world.time < last_emit_neurogas + DEFILER_GAS_COOLDOWN) //Sure, let's use this.
+		to_chat(src, "<span class='xenodanger'>You are not ready to emit neurogas again. This ability will be ready in [(last_emit_neurogas + DEFILER_GAS_COOLDOWN - world.time) * 0.1] seconds.</span>")
+		return FALSE
+
+	if(stagger)
+		to_chat(src, "<span class='xenowarning'>You try to emit neurogas but are staggered!</span>")
+		return
+
+	if(!check_plasma(200))
+		return
+
+	last_emit_neurogas = world.time
+	use_plasma(200)
+
+	//give them fair warning
+	visible_message("<span class='danger'>Tufts of smoke begin to billow from [src]!</span>", \
+	"<span class='xenodanger'>Your dorsal vents widen, preparing to emit neurogas. Keep still!</span>")
+
+	spawn(DEFILER_GAS_COOLDOWN)
+		playsound(loc, 'sound/effects/xeno_newlarva.ogg', 50, 0)
+		to_chat(src, "<span class='xenodanger'>You feel your dorsal vents bristle with neurotoxic gas. You can use Emit Neurogas again.</span>")
+		update_action_button_icons()
+
+	emitting_gas = TRUE //We gain bump movement immunity while we're emitting gas.
+	if(!do_after(src, DEFILER_GAS_CHANNEL_TIME, TRUE, 5, BUSY_ICON_HOSTILE))
+		emitting_gas = FALSE
+		return
+	emitting_gas = FALSE
+
+	if(stagger) //If we got staggered, return
+		to_chat(src, "<span class='xenowarning'>You try to emit neurogas but are staggered!</span>")
+		return
+
+	round_statistics.defiler_neurogas_uses++
+
+	visible_message("<span class='xenodanger'>[src] emits a noxious gas!</span>", \
+	"<span class='xenodanger'>You emit neurogas!</span>")
+	dispense_gas(3)
+
+/mob/living/carbon/Xenomorph/Defiler/proc/dispense_gas(count = 0)
+	while(count)
```

Comment:
Make sure to `set waitfor = FALSE` if you don't want to this to lock the other procs, but make sure the caller does not require a return from the proc.
`count = 0` does no longer seem necessary as a passed argument, now does it?

Reply:
It might be github, but is this properly tabbed?